src/resource.c

```diff
@@ -735,6 +735,15 @@ static const uint16_t cache_ignore_options[] = { COAP_OPTION_ETAG };
 
   assert( session );
 
+  /* Check if there is already maximum number or subscribers present */
+#if (COAP_RESOURCE_MAX_SUBSCRIBER > 0)
+  uint8_t subscriber_count = 0;
+  LL_COUNT(resource->subscribers, s, subscriber_count);
+  if (subscriber_count > COAP_RESOURCE_MAX_SUBSCRIBER) {
+    return NULL; /* Signal error */
```

Comment:
What happens if a client is already registered and wants to check the subscription is already there for doing a refresh (i.e. client has just rebooted)?

I think this code should be a few lines further down after the `coap_find_observer()` test.